src/components/application_manager/src/policies/policy_handler.cc

```diff
@@ -154,26 +155,32 @@ typedef std::set<application_manager::ApplicationSharedPtr,
 
 struct DeactivateApplication {
   explicit DeactivateApplication(
-      const connection_handler::DeviceHandle& device_id)
-      : device_id_(device_id) {}
+      const connection_handler::DeviceHandle& device_id,
+            StateController& state_ctrl)
+      : device_id_(device_id), state_ctrl_(state_ctrl) {}
 
   void operator()(const ApplicationSharedPtr& app) {
     if (device_id_ == app->device()) {
-      ApplicationManagerImpl::instance()->SetState<true>(
-          app->app_id(),
+      state_ctrl_.SetRegularState(
+          app,
           mobile_apis::HMILevel::HMI_NONE,
-          mobile_apis::AudioStreamingState::NOT_AUDIBLE);
+          mobile_apis::AudioStreamingState::NOT_AUDIBLE,
+                  true);
     }
   }
 
  private:
   connection_handler::DeviceHandle device_id_;
+  StateController& state_ctrl_;
 };
 
 struct SDLAlowedNotification {
```

Comment:
@Kozoriz May be change structure name to correct one? SDLAlowedNotification -> SDLAllowedNotification